### PR TITLE
Various Issues - Tidying up & finishing off all the screens

### PR DESCRIPTION
**Pull request by fluffyemily**
#45 #46 #47 #48 #76 #82 #80 #88 #89 #90

### CODE REVIEW

**Comment by fluffyemily** [comment]
[I want to receive an actionable overview of relevant information about places near me, so that my day is not ruined with FOMO](https://trello.com/c/VPWm03rY/49-i-want-to-receive-an-actionable-overview-of-relevant-information-about-places-near-me-so-that-my-day-is-not-ruined-with-fomo)

**Comment by fluffyemily** [comment]
[As a user I want to be able to visit the website of the place I am looking at so that I can find out more about it.](https://trello.com/c/zNEuTpBe/168-as-a-user-i-want-to-be-able-to-visit-the-website-of-the-place-i-am-looking-at-so-that-i-can-find-out-more-about-it)

**Comment by fluffyemily** [comment]
[As a user I want to view the directions to a place so that I can easily decide how to get there.](https://trello.com/c/q0qJbEj7/167-as-a-user-i-want-to-view-the-directions-to-a-place-so-that-i-can-easily-decide-how-to-get-there)

**Comment by fluffyemily** [comment]
[As a user I want to read the reviews for a place so that I can make better decisions about what I want to do next](https://trello.com/c/rJvuwaeI/169-as-a-user-i-want-to-read-the-reviews-for-a-place-so-that-i-can-make-better-decisions-about-what-i-want-to-do-next)

File: Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift [hunk 1]
```diff
@@ -163,6 +152,15 @@ class PlaceCarouselViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
+        if let backgroundImage = UIImage(named: "map_background") {
```
Comment (mcomella) [inline Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift]: This image isn't large enough, and tiles, for plus-size devices.

File: Prox/Prox/PlaceDetails/PlaceDetailsCardViewController.swift [hunk 2]
```diff
@@ -101,9 +102,68 @@ class PlaceDetailsCardViewController: UIViewController {
...
         cardView.updateUI(forPlace: place)
+        setLocation(location: locationProvider?.getCurrentLocation())
+        setupCardInteractions()
+
         pageControl.numberOfPages = place.photoURLs?.count ?? 0
     }
 
+    private func setLocation(location: CLLocation?) {
+        if let lastTravelTimes = place.lastTravelTime {
+            self.cardView.updateTravelTimesUI(travelTimes: lastTravelTimes)
```
Comment (mcomella) [inline Prox/Prox/PlaceDetails/PlaceDetailsCardViewController.swift]: There should be a return here.

File: Prox/Prox/PlaceDetails/PlaceDetailsCardViewController.swift [hunk 3]
```diff
@@ -101,9 +102,68 @@ class PlaceDetailsCardViewController: UIViewController {
...
         }
         cardView.updateUI(forPlace: place)
+        setLocation(location: locationProvider?.getCurrentLocation())
+        setupCardInteractions()
+
         pageControl.numberOfPages = place.photoURLs?.count ?? 0
     }
 
+    private func setLocation(location: CLLocation?) {
+        if let lastTravelTimes = place.lastTravelTime {
```
Comment (mcomella) [inline Prox/Prox/PlaceDetails/PlaceDetailsCardViewController.swift]: We should add a comment here (and in the similar code) that if we're not cached and a request is in motion, we're not waiting for a response – we're creating a new request.

**Comment by mcomella** [comment]
Filed #102 and #103.

**Comment by mcomella** [comment]
I rebased the changes onto master. Ricardo added some assets which conflicted, so I removed the conflicting commit and the assets seem to be the most up-to-date state.

File: Prox/Prox/PlaceDetails/PlaceDetailsCardView.swift [hunk 4]
```diff
@@ -115,6 +115,7 @@ class PlaceDetailsCardView: UIView {
     lazy var tripAdvisorReviewView: ReviewContainerView = {
         let view = ReviewContainerView(color: Colors.tripAdvisor, mode: .detailsView)
         view.reviewSiteLogo.image = UIImage(named: "logo_ta")
+        view.isUserInteractionEnabled = true
```
Comment (fluffyemily) [inline Prox/Prox/PlaceDetails/PlaceDetailsCardView.swift]: no, actually this should not be on this view at all. I think I removed it from the label and left it here when it should have been the other way around

File: Prox/Prox/PlaceDetails/PlaceDetailsCardViewController.swift [hunk 5]
```diff
@@ -122,6 +119,47 @@ class PlaceDetailsCardViewController: UIViewController {
...
     }
 
+    private func setupCardInteractions() {
+        cardView.urlLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(openPlaceURL(gestureRecgonizer:))))
+        cardView.yelpReviewView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(openYelpReview(gestureRecgonizer:))))
+        cardView.tripAdvisorReviewView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(openTripAdvisorReview(gestureRecgonizer:))))
+        cardView.travelTimeView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(openDirections(gestureRecgonizer:))))
+    }
+
+    @objc private func openPlaceURL(gestureRecgonizer: UITapGestureRecognizer) {
```
Comment (fluffyemily) [inline Prox/Prox/PlaceDetails/PlaceDetailsCardViewController.swift]: What broke was me removing the `isUserinteractionEnabled = true` flag from the URL label which prevented the `UITapGestureRecognizer` from being fired. This code works just fine.

**Pull request merged**